rtsp.go

```diff
@@ -79,6 +83,8 @@ type rtspCamera struct {
 
 	latestFrame atomic.Pointer[image.Image]
 
+	mustUseH264 bool
```

Comment:
In the future we can pass in the codecInfo for multiple encodings to `reconnectClient` and have it fail if it's not the same codec, what you've done is fine for now, also lets us not keep track of as much state in this struct.

Reply:
> pass in the codecInfo for multiple encodings to reconnectClient

But isn't `codecInfo` handled inside `reconnectClient`

Reply:
Yes agree, would be nice to not add to `rtspCamera` struct here, but this seems good for now since we only have two models.

For "stateless" option, we could:
- Add codec whitelist param to `reconnectClient`.
- Pass codec whitelist to `clientReconnectBackgroundWorker` based on model.

Reply:
Would it be Viam `Act Decisively` of me to just go ahead and add model `rtsp-h265` and implement a more elegant solution with enums.

Reply:
I'm gonna do it lol